src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -676,7 +676,7 @@ MulticopterAttitudeControl::publish_rates_setpoint()
 void
 MulticopterAttitudeControl::publish_rate_controller_status()
 {
-	rate_ctrl_status_s rate_ctrl_status;
+	rate_ctrl_status_s rate_ctrl_status = {};
```

Comment:
The problem with this is that we do this a lot in code and end up spending a significant amount of CPU in initializers that get overridden a few lines down (according to profiling that I did a year or two back). We need a way to avoid the double initialization within a few lines. Using C-style curly braces initializers would be an option.

Reply:
I'd really hope that the compiler can see that we write the same memory twice and skip the first one. Inline initialization would of course be best, but it is very easy to introduce bugs there if fields are re-ordered, and very difficult to diagnose since the names are in 2 different files with only an implicit coupling via the order. Constructors are of course problematic because the struct/class is no longer POD.

My personal preference would be to initialize all floating point values with NaN (C++11-style in the definitions) as part of the message generation code, then let the compiler do the rest.

Reply:
Tested with godbolt on gcc-arm-none, all compile to the same thing:
https://godbolt.org/z/lEg29M

Reply:
Cool, good to see that my profiling testing is outdated then.

Reply:
The 'safe' version with all initialized in the declaration also optimizes away, unless you forget an element, then it blows up a bit:
https://godbolt.org/z/dk4AHx

Reply:
Ok, that is exactly what I meant. I'm looking for a C++ construct to force all elements. Because people add one into the message, other code doesn't get updated -> as the result many are not initialized and so the default is that it blows up.

Reply:
> Cool, good to see that my profiling testing is outdated then.

We've also switched to an optimized memset in NuttX since then. We've also found new tools (https://github.com/orbcode/orbuculum).